File: Intro_sockets/server/models/band_model.js
const { v4: uuidV4 } = require('uuid');

class BandModel {
    constructor(name = 'no-name') {
        this.id = uuidV4();
        this.name = name;
        this.votes = 0;
    }


    static fromMap(map) {
        return new BandModel(map.name);
    }
}
module.exports = BandModel;

File: Intro_sockets/intro_sockets_app/lib/models/band_model.dart
class BandModel {
  String? id;
  String? name;
  int? votes;

  BandModel({
    this.id,
    this.name,
    this.votes,
  });

  factory BandModel.fromMap(Map<String,dynamic> data){
    return BandModel(
      id: data['id'],
      name: data['name'],
      votes: data['votes'],
    );
  }

  Map<String,dynamic> toMap(){
    return {
      'id': id,
      'name': name,
      'votes': votes,
    };
  }
}
